### PR TITLE
PACT-656 Added the missing style classes on error for start and end d…

### PR DESCRIPTION
**Pull request by userjd**
…ate edit record fields

### CODE REVIEW

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html [hunk 1]
```diff
@@ -99,6 +99,24 @@
     </div>
 }
 
+@dateFieldSet(datePrefix: String, dateLabel: String) = {
+
+    @editSetRecordForm.error(datePrefix+"-date-day") match {
+        case Some(_) => {
```
Comment (tna-erasmos) [inline app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:105]: You could remove the braces for both `Some` and `None`, so use a single line for each.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html [hunk 2]
```diff
@@ -99,6 +99,24 @@
...
+                <div class="govuk-form-group  govuk-form-group--error">}
+        case None => {
+        <div class="govuk-form-group">}
+            }
+            @govukFieldset(
+                Fieldset(
```
Comment (tna-erasmos) [inline app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:111]: I'd indent the field names.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html [hunk 3]
```diff
@@ -112,34 +130,43 @@
...
     <div class="govuk-date-input" id="@prefix">
         <div class="govuk-date-input__item">
-            <div class="govuk-form-group">
             @govukInput(Input(
                 id = prefix + "-date-day",
                 name = prefix + "-date-day",
                 value = editSetRecordForm.data.get(prefix + "-date-day"),
                 label = Label(classes = "govuk-label govuk-date-input__label", forAttr = Some("@{prefix}-date-day"), content = Text(messages("edit-set.record.edit.dates.day"))),
-                classes = "govuk-input govuk-date-input__input govuk-input--width-2"))
+                classes = editSetRecordForm.error(prefix+"-date-day") match {
```
Comment (tna-erasmos) [inline app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:138]: I'd suggest a more succinct approach, with less duplication:

```
classes = "govuk-input govuk-date-input__input govuk-input--width-2" + editSetRecordForm.error(prefix+"-date-day").map(_ => " govuk-input--error").getOrElse("") 
```

This applies to the other two cases, of course.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html [hunk 4]
```diff
@@ -112,34 +130,43 @@
...
-            <div class="govuk-form-group">
             @govukInput(Input(
                 id = prefix + "-date-month",
                 name = prefix + "-date-month",
                 value = editSetRecordForm.data.get(prefix + "-date-month"),
                 label = Label(classes = "govuk-label govuk-date-input__label", forAttr = Some("@{prefix}-date-month"), content = Text(messages("edit-set.record.edit.dates.month"))),
-                classes = "govuk-input govuk-date-input__input govuk-input--width-2"))
-            </div>
+                classes = editSetRecordForm.error(prefix+"-date-day") match {
+                                case Some(_) => {
```
Comment (tna-erasmos) [inline app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:152]: See recommendation above.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html [hunk 5]
```diff
@@ -112,34 +130,43 @@
...
-            <div class="govuk-form-group">
             @govukInput(Input(
                 id = prefix + "-date-year",
                 name = prefix + "-date-year",
                 value = editSetRecordForm.data.get(prefix + "-date-year"),
                 label = Label(classes = "govuk-label govuk-date-input__label", forAttr = Some("@{prefix}-date-year"), content = Text(messages("edit-set.record.edit.dates.year"))),
-                classes = "govuk-input govuk-date-input__input govuk-input--width-4"))
-            </div>
+                classes = editSetRecordForm.error(prefix+"-date-day") match {
+                                case Some(_) => {
```
Comment (tna-erasmos) [inline app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:165]: See recommendation above.

**Review by tna-erasmos — APPROVED** [review]
LGTM. I did add a few suggestions, however.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 1 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html]: Without the braces, there's an error compiling template and doesn't recognise `case`

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 2 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:109]: Done

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 3 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html]: Done

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 4 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html]: Done

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 5 above)
Comment (userjd) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html]: Done

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 1 above)
Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html]: I meant on line 105 - but it's OK, as you've followed my suggested approach.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 2 above)
Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:109]: Confirmed.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 3 above)
Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html]: Confirmed.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 4 above)
Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html]: Confirmed.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 5 above)
Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html]: Confirmed.

**Review by rwalpole — APPROVED** [review]

**Pull request merged**